service/history/handler.go

```diff
@@ -528,7 +528,9 @@ func (h *Handler) StartWorkflowExecution(ctx context.Context, request *historyse
 	if err != nil {
 		return nil, h.convertError(err)
 	}
-	response.Clock, err = shardContext.NewVectorClock()
+	if response.Clock == nil {
+		response.Clock, err = shardContext.NewVectorClock()
+	}
 	if err != nil {
 		return nil, h.convertError(err)
 	}
```

Comment:
This error check needs to be inside `if` block.

Reply:
It probably doesn't matter much but you're right.